diff --git a/commerce_recurring.rules.inc b/commerce_recurring.rules.inc
index 5fe21bf..aabeb84 100644
--- a/commerce_recurring.rules.inc
+++ b/commerce_recurring.rules.inc
@@ -202,9 +202,9 @@ function commerce_recurring_rules_action_info() {
       ),
     ) + $profiles,
     'provides' => array(
-      'commerce_order' => array(
+      'new_order' => array(
         'type' => 'commerce_order',
-        'label' => t('Commerce Order'),
+        'label' => t('New Order'),
       ),
     ),
     'group' => t('Commerce recurring'),
@@ -527,7 +527,7 @@ function commerce_recurring_rules_generate_order_from_recurring() {
   }
   $recurring_wrapper->commerce_recurring_order[] = $order->order_id;
   entity_save('commerce_recurring', $recurring_wrapper->value());
-  return array('commerce_order' => $order);
+  return array('new_order' => $order);
 }
 
 /**
